DataFormats/HcalDigi/interface/QIE10DataFrame.h

```diff
@@ -0,0 +1,55 @@
+#ifndef DATAFORMATS_HCALDIGI_QIE10DATAFRAME_H
+#define DATAFORMATS_HCALDIGI_QIE10DATAFRAME_H
+
+#include "DataFormats/HcalDetId/interface/HcalDetId.h"
+#include "DataFormats/Common/interface/DataFrame.h"
+#include <ostream>
+
+/** Precision readout digi from QIE10 including TDC information
+
+ */
+class QIE10DataFrame : protected edm::DataFrame {
+public:
+
+  static const int WORDS_PER_SAMPLE = 2;
+  static const int HEADER_WORDS = 1;
+
+  QIE10DataFrame() { }
+  QIE10DataFrame(const edm::DataFrameContainer& c, edm::DataFrame::size_type i) : edm::DataFrame(c,i) { }
+  QIE10DataFrame(edm::DataFrame df) : edm::DataFrame(df) { }
+
+  class Sample {
+  public:
+    Sample(const edm::DataFrame& frame, edm::DataFrame::size_type i) : frame_(frame),i_(i) { }
+    int adc() const { return frame_[i_]&0xFF; }
```

Comment:
Please add masks and offsets constants to be used in these methods; in order to avoid magic numbers scattered in many places.
[in a future PR]